butterknife-compiler/src/main/java/butterknife/compiler/BindingClass.java

```diff
@@ -350,8 +350,12 @@ private MethodSpec createNewBindToTargetMethod() {
     }
 
     if (bindNeedsFinder()) {
-      result.addParameter(FINDER, "finder")
-          .addParameter(Object.class, "source");
+      if (hasMethodBindings()) {
```

Comment:
Actually don't worry about it. I'm making other changes to generated code and I'll rebase your change on top of it (and add this extra functionality) once I'm done.